src/compiler.rs

```diff
@@ -132,4 +137,36 @@ impl Compiler {
             file: out_file,
         }
     }
+
+    pub fn skip_compile(&self, out_file: String) -> CompileOutput {
+        CompileOutput {
+            status: ExitStatusExt::from_raw(0),
+            stdout: vec![],
+            stderr: vec![],
+            file: out_file,
+        }
+    }
+}
+
+pub fn should_compile(in_files: &[String], out_file: &str) -> bool {
```

Comment:
I feel like `should_compile` is a bit too general. Maybe rename it to something like `is_source_modified`? What do you think?

Reply:
That sounds better actually.